src/main/java/org/openjdk/jextract/impl/Parser.java

```diff
@@ -60,49 +104,21 @@ public Declaration.Scoped parse(Path path, Collection<String> args) {
                 },
             true, args.toArray(new String[0])) ;
             MacroParserImpl macroParser = MacroParserImpl.make(treeMaker, logger, tu, args)) {
+            return collectDeclarations(tu, macroParser);
+        }
+    }
 
-            List<Declaration> decls = new ArrayList<>();
-            Cursor tuCursor = tu.getCursor();
-            tuCursor.forEach(c -> {
-                SourceLocation loc = c.getSourceLocation();
-                if (loc == null) {
-                    return;
-                }
-
-                SourceLocation.Location src = loc.getFileLocation();
-                if (src == null) {
-                    return;
-                }
-
-
-                if (c.isDeclaration()) {
-                    if (c.kind() == CursorKind.UnexposedDecl ||
-                            c.kind() == CursorKind.Namespace) {
-                        c.forEach(t -> {
-                            Declaration declaration = treeMaker.createTree(t);
-                            if (declaration != null) {
-                                decls.add(declaration);
-                            }
-                        });
-                    } else {
-                        Declaration decl = treeMaker.createTree(c);
-                        if (decl != null) {
-                            decls.add(decl);
-                        }
-                    }
-                } else if (isMacro(c) && src.path() != null) {
-                    SourceRange range = c.getExtent();
-                    String[] tokens = c.getTranslationUnit().tokens(range);
-                    Optional<Declaration.Constant> constant = macroParser.parseConstant(c, c.spelling(), tokens);
-                    if (constant.isPresent()) {
-                        decls.add(constant.get());
+    public Declaration.Scoped parse(String name, String content, Collection<String> args) {
```

Comment:
The only use of this method seems to be in `JextractTool`, so I think the old version can just be removed?

Reply:
Will remove it.